### DISCUSSION TITLE
How can the state transition from LISTEN -> SYN_SENT be occurred?

### DISCUSSION THREAD

**Discussion by park-sunho** [ANLAB-KAIST/KENSv3#110] [Project Q&A 2. Transport Layer (TCP)] [answered]
In the TCP state transition diagram, I saw LISTEN -> SYN_SENT can be triggered if the application sends data using the socket in LISTEN state. But I cannot get the idea on how user application sends data using listening socket. Since listening socket does not have destination ip and port, I thought we cannot send data using the socket in TCP. Can you give me any idea on this?

Thank you in advance.

**Comment by whkim0** [comment, marked as answer]
State `LISTEN` in the TCP state transition diagram is not equivalent to the `listen()` system call. Hope that this answered your question.
